### PR TITLE
Rename valid test folders to be consistent with naming convention of invalid test folders.

### PR DESCRIPTION
**Pull request by qingling128**
Follow up of https://github.com/GoogleCloudPlatform/ops-agent/pull/111 as we decided to hold on renaming in that PR to avoid rebase conflicts before GA.

To summarize the renaming in this PR:

```
$ $ git diff origin/master..HEAD  | grep rename | sed 's/confgenerator\/testdata\/valid\///g;s/\/input.yaml//g' | grep -v golden
rename from linux/logging-no_conf
rename to linux/logging-default_overrides_disable_all
rename from linux/logging-complex_logs
rename to linux/logging-pipeline_multiple_pipelines
rename from linux/logging-complex_logs_with_processors
rename to linux/logging-processor_parse_json_and_parse_regex_types
rename from linux/logging-processor_parse_regex_on_default_pipeline
rename to linux/logging-processor_parse_regex_type_on_default_pipeline
rename from linux/logging-multiple_file_sources
rename to linux/logging-receiver_files_type_multiple_receivers
rename from linux/logging-multiple_syslog_sources
rename to linux/logging-receiver_syslog_type_multiple_receivers
rename from linux/metrics-no_conf
rename to linux/metrics-default_overrides_disable_all
rename from linux/metrics-exclude_metrics_by_prefixes
rename to linux/metrics-processor_exclude_metrics_type_filter_by_prefixes
rename from linux/metrics-exclude_metrics_by_prefixes_with_glob
rename to linux/metrics-processor_exclude_metrics_type_filter_with_globs
rename from linux/metrics-custom_collection_interval
rename to linux/metrics-receiver_custom_collection_interval
rename from windows/logging-no_conf
rename to windows/logging-default_overrides_disable_all
rename from windows/logging-multiple_file_sources
rename to windows/logging-receiver_files_type_multiple_receivers
rename from windows/metrics-no_conf
rename to windows/metrics-default_overrides_disable_all
rename from windows/metrics-turn_off_iis
rename to windows/metrics-default_overrides_disable_iis
rename from windows/metrics-turn_off_mssql
rename to windows/metrics-default_overrides_disable_mssql
rename from windows/valid_config
rename to windows/metrics-pipeline_multiple_pipelines
rename from windows/metrics-custom_collection_interval
rename to windows/metrics-receiver_custom_collection_interval
```

It also deletes a test case

```
linux/logging-multiple_google_exporters
```

and adds a few test cases:

```
linux/all-backward_compatible_with_explicit_exporters
windows/all-backward_compatible_with_explicit_exporters
windows/metrics-processor_exclude_metrics_type_filter_by_prefixes
windows/metrics-processor_exclude_metrics_type_filter_with_globs
```

### CODE REVIEW

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
I wish I could comment on the individual files. Sigh…

> `linux/logging-multiple_file_sources` → `linux/logging-exporter_google_type_multiple_instances`
`windows/logging-multiple_file_sources` → `windows/logging-receiver_files_type_multiple_receivers`
`linux/logging-multiple_google_exporters` → `linux/logging-receiver_files_type_multiple_receivers`

These three are inconsistent. You probably want to use `logging-receiver_files_type_multiple_receivers` for the `logging-multiple_file_sources` configs, and the `logging-multiple_google_exporters` one may just need to go away, given that it's identical to the `logging-multiple_file_sources` one…

> `linux/metrics-exclude_metrics_by_prefixes` → `linux/metrics-processor_exclude_metrics_type_filter_by_prefixes`
`linux/metrics-exclude_metrics_by_prefixes_with_glob` → `linux/metrics-processor_exclude_metrics_type_filter_with_globs`
`windows/metrics-turn_off_iis` → `windows/metrics-processor_exclude_metrics_type_disable_iis`
`windows/metrics-turn_off_mssql` → `windows/metrics-processor_exclude_metrics_type_disable_mssql`

Any reason we don't have consistent naming for these? Or do we need the equivalent of the first two for Windows in addition to the "disable IIS" and "disable MSSQL" cases?

> `windows/valid_config` → `windows/metrics-pipeline_multiple_instances`

Do we need one of these for Linux? Also, shouldn't we remove the `exporters:` key?

**Comment by qingling128** [comment]
Addressed the feedback. PTAL

**Comment by qingling128** [comment]
Ready for review again.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
Actually, just realized — you've renamed the `default_config` test, but we reference that test by name in `confgenerator_test.go`. Don't you need to update that?

**Comment by qingling128** [comment]
Just fixed the builtInConfTestName. PTAL

File: confgenerator/confgenerator_test.go [hunk 1]
```diff
@@ -30,7 +30,10 @@ import (
 const (
 	validTestdataDir   = "testdata/valid"
 	invalidTestdataDir = "testdata/invalid"
-	defaultConfig      = "default-config.yaml"
+	// Relative path to the ops-agent/confgenerator folder.
+	defaultUserConfPath = "default-config.yaml"
```
Comment (igorpeshansky) [inline confgenerator/confgenerator_test.go:34]: [Optional] I wouldn't bother renaming this…

File: confgenerator/confgenerator_test.go [hunk 2]
```diff
@@ -30,7 +30,10 @@ import (
 const (
 	validTestdataDir   = "testdata/valid"
 	invalidTestdataDir = "testdata/invalid"
-	defaultConfig      = "default-config.yaml"
+	// Relative path to the ops-agent/confgenerator folder.
```
Comment (igorpeshansky) [inline confgenerator/confgenerator_test.go:33]: The above paths are also relative to the `confgenerator` folder. Worth moving the comment up?

File: confgenerator/confgenerator_test.go [hunk 3]
```diff
@@ -30,7 +30,10 @@ import (
...
+	defaultUserConfPath = "default-config.yaml"
+	// Test name inside ops-agent/confgenerator/testdata/valid/{linux|windows}
```
Comment (igorpeshansky) [inline confgenerator/confgenerator_test.go:35]: I would not mention the `ops-agent/` prefix…

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
Minor things.

File: confgenerator/confgenerator_test.go (hunk 1 above)
Comment (qingling128) [reply confgenerator/confgenerator_test.go:34]: The concern was that now we have split the built-in config and user specified config, `default-config.yaml` might not be descriptive enough.

File: confgenerator/confgenerator_test.go (hunk 3 above)
Comment (qingling128) [reply confgenerator/confgenerator_test.go]: Removed

**Comment by qingling128** [comment]
Updated. PTAL

File: confgenerator/confgenerator_test.go (hunk 2 above)
Comment (qingling128) [reply confgenerator/confgenerator_test.go]: Yup. Moved it up.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Pull request merged**